src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -210,11 +212,18 @@ async def setup_pytest_for_target(
             coverage_args.extend(["--cov", package])
 
     specified_source_file_names = sorted(specified_source_files.snapshot.files)
+    results_dir = test_options.values.results_dir
+    results_file_path = (
+        (Path(results_dir) / f"{config.address.path_safe_spec}.xml").absolute().as_posix()
```

Comment:
Absolute paths aren't safe to use from rules... sorry: we need to do a better job of sandboxing so that that is clearer.

Instead, you should use a relative path, which will end up inside the snapshot that is captured by the test run, and then mark the file/directory as captured here: https://github.com/pantsbuild/pants/blob/79b966ce77f746995e357ee06d14d1888f20d16e/src/python/pants/backend/python/rules/pytest_runner.py#L239

Finally, to actually publish the file into the `dist` dir at the end, you'd want to update https://github.com/pantsbuild/pants/blob/79b966ce77f746995e357ee06d14d1888f20d16e/src/python/pants/core/goals/test.py#L45 to include an optional test report XML name.

Reply:
One thing I can't figure out... is how to work w/Digest
How do I distinguish the coverage data from my data.
I see that coverage just consumes the entire digest:
https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/rules/pytest_runner.py#L248

Not sure how to handle multiple files/directories that are generated by the EPR..